modules/oe_theme_helper/tests/src/Functional/Plugin/field_group/FieldInPageNavigationFormatterTest.php

```diff
@@ -0,0 +1,284 @@
+      'weight' => 60,
+      'label' => 'Group 6',
+      'format_type' => 'html_element',
+      'format_settings' => [
+        'show_label' => FALSE,
+      ],
+    ];
+    $group_6 = $this->createGroup('node', 'test', 'view', 'default', $data);
+    $data = [
+      'label' => 'In-page navigation group',
+      'weight' => 0,
+      'group_name' => 'group_inpage_nav',
+      'children' => [
+        $group_1->group_name,
+        $group_2->group_name,
+        $group_3->group_name,
+        $group_4->group_name,
+        $group_5->group_name,
+        $group_6->group_name,
+        'field_test_5',
+        'field_test_6',
+      ],
+      'format_type' => 'oe_theme_helper_in_page_navigation',
+    ];
+    $this->createGroup('node', 'test', 'view', 'default', $data);
+    // Create values for the testing node.
+    $node_values = ['type' => "test"];
+    foreach ($this->fields as $field_name => $field) {
+      if ($field['content'] !== '') {
+        $node_values[$field_name][0]['value'] = $field['content'];
+      }
+    }
+    $node = $this->drupalCreateNode($node_values);
+    $html = $this->drupalGet('node/' . $node->id());
+    $crawler = new Crawler($html);
+    // The content is break into two elements, for navigation and content.
+    $navigation_selector = 'article .ecl-container .ecl-col-lg-3 .ecl-inpage-navigation';
+    $content_selector = 'article .ecl-container .ecl-col-lg-9';
+    $this->assertSession()->elementExists('css', $navigation_selector);
+    $this->assertSession()->elementExists('css', $content_selector);
+    // The navigation links match the content anchors.
+    $anchor_links = [
+      [
+        'id' => 'inline-nav-1',
+        'label' => 'Group 1',
+        'selector' => 'summary',
+      ], [
+        // If a group has an id configured in the group configuration,
+        // use that as anchor.
+        'id' => 'group_2_id',
+        'label' => 'Group 2',
+        'selector' => 'summary',
+      ], [
+        'id' => 'inline-nav-2',
+        'label' => 'Group 4',
+        'selector' => 'summary',
+      ], [
+        'id' => 'inline-nav-3',
+        'label' => 'Field 5 label',
+        'selector' => 'div',
+      ],
+    ];
+    foreach ($anchor_links as $key => $value) {
+      $element = $crawler->filter($navigation_selector . " #ecl-inpage-navigation-list  [href='#" . $value['id'] . "']");
+      // Link label.
+      $this->assertEquals($value['label'], trim($element->text()));
+      // Content label.
+      $this->assertSession()->elementTextContains('css', $content_selector . " #" . $value['id'] . " > " . $value['selector'], $value['label']);
+    }
+    // If a group has no fields, don't include it in the navigation.
+    $this->assertSession()->pageTextNotContains('Group 5');
+    // If a group has no visible label, don't include it in the navigation.
+    $this->assertSession()->pageTextNotContains('Group 6');
+    // If a field has no visible label, don't include it in the navigation.
```

Comment:
If a field has label but hidden, don't include it in the navigation.

Reply:
Updated